src/react_native/core.cljs

```diff
@@ -134,26 +133,50 @@
 
 (def use-context react/useContext)
 
+(defn use-ref-atom
+  [value]
+  (let [ref (use-ref (atom value))]
+    (.-current ^js ref)))
+
+(defn get-js-deps
+  [deps]
+  (if deps
+    (let [prev-state (use-ref-atom {:value false :deps nil})
+          prev-deps  (:deps @prev-state)
+          prev-value (:value @prev-state)]
+      (if (and (not (nil? prev-deps)) (not= (count deps) (count prev-deps)))
+        (throw (js/Error. "Hooks can't have a different number of dependencies across re-renders"))
```

Comment:
If this error is thrown it's not gonna be easy to debug because there could be too many views that could fail for this reason. We should at least log the dependencies themselves to help the dev pinpoint what failed. I think it's best too if we use `ex-info` because it's more idiomatic and there's no need to limit ourselves to an error string.

Reply:
agree, we can do it, but i don't think this will ever happen

Reply:
Good point, it should be pretty rare.